File: app/src/main/java/org/thoughtcrime/securesms/conversationlist/model/MessageResult.java
package org.thoughtcrime.securesms.conversationlist.model;

import androidx.annotation.NonNull;

import org.thoughtcrime.securesms.recipients.Recipient;

/**
 * Represents a search result for a message.
 */
public class MessageResult {

  public final Recipient conversationRecipient;
  public final Recipient messageRecipient;
  public final String    body;
  public final String    bodySnippet;
  public final long      threadId;
  public final long      messageId;
  public final long      receivedTimestampMs;
  public final boolean   isMms;

  public MessageResult(@NonNull Recipient conversationRecipient,
                       @NonNull Recipient messageRecipient,
                       @NonNull String body,
                       @NonNull String bodySnippet,
                       long threadId,
                       long messageId,
                       long receivedTimestampMs,
                       boolean isMms)
  {
    this.conversationRecipient = conversationRecipient;
    this.messageRecipient      = messageRecipient;
    this.body                  = body;
    this.bodySnippet           = bodySnippet;
    this.threadId              = threadId;
    this.messageId             = messageId;
    this.receivedTimestampMs   = receivedTimestampMs;
    this.isMms                 = isMms;
  }
}

File: app/src/main/java/org/thoughtcrime/securesms/search/MessageResult.kt
package org.thoughtcrime.securesms.search

import org.thoughtcrime.securesms.recipients.Recipient

/**
 * Represents a search result for a message.
 */
data class MessageResult(
  val conversationRecipient: Recipient,
  val messageRecipient: Recipient,
  val body: CharSequence,
  val bodySnippet: CharSequence,
  val threadId: Long,
  val messageId: Long,
  val receivedTimestampMs: Long,
  val isMms: Boolean
)
